common/v2/services/UnstoppableService/functions.ts

```diff
@@ -0,0 +1,30 @@
+import UnstoppableResolution from './unstoppableResolution';
+import { ResolutionErrorCode, ResolutionError } from '@unstoppabledomains/resolution';
+
+export const getResolvedAddress = async (name: string, asset: string): Promise<string> => {
+  const provider = new UnstoppableResolution();
+  return await provider.getResolvedAddress(name, asset);
+};
+
+export const parseResolutionError = (err: ResolutionError, callback: any, params?: any) => {
+  const setResolutionError = callback;
+  const { ticker, name } = params;
+  switch (err.code) {
+    case ResolutionErrorCode.UnregisteredDomain: {
+      setResolutionError({ code: err.code, params: { $domain: name } });
+      break;
+    }
+    case ResolutionErrorCode.UnsupportedDomain: {
+      setResolutionError({ code: err.code, params: { $domain: name } });
+      break;
+    }
+    case ResolutionErrorCode.RecordNotFound: {
+      setResolutionError({ code: err.code, params: { $recordName: ticker, $domain: name } });
+      break;
+    }
+    case ResolutionErrorCode.UnspecifiedCurrency: {
+      setResolutionError({ code: err.code, params: { $domain: name, $currencyTicker: ticker } });
```

Comment:
Why all of that need to be so custom?
I would use `setResolutionError({ code: err.code, params: { $domain: name,  $recordName: ticker,  $currencyTicker: ticker } }); } });` just for everything.